src/main/java/it/gov/pagopa/pu/bff/service/organization/OrganizationRetrieverServiceImpl.java

```diff
@@ -98,4 +100,12 @@ private List<DebtPositionTypeOrgCountByOrganizationId> getDebtPositionTypeOrgCou
     return collection.getEmbedded().getDebtPositionTypeOrgCountByOrganizationIds();
   }
 
+  public String getOrgFiscalCode(Long organizationId, UserInfo loggedUser, String accessToken) {
+    Organization organization = organizationService.getOrganizationByOrganizationId(organizationId, accessToken);
+    if(organization!=null && organization.getBrokerId()!=null && organization.getBrokerId().equals(loggedUser.getBrokerId())){
+      return organization.getOrgFiscalCode();
+    }else{
+      throw new ResourceNotFoundException("Organization having organizationId "+ organizationId +" not found");
```

Comment:
if it exists, but under another broker, this exception could be ambigous to debug the error...
could we add a log o introduce a new exception for this case?